xsuite/src/world/signer.ts

```diff
@@ -56,26 +56,30 @@ export class KeystoreSigner extends UserSigner {
 export class Keystore {
   data: any;
   password: string;
+  kind: "mnemonic" | "secretKey";
 
   constructor(data: any, password: string) {
     this.data = data;
     this.password = password;
+    this.kind = data.kind ?? "mnemonic";
```

Comment:
The default value should be "secretKey". Also, it would be great to validate the `kind` field and fail if the field is not "mnemonic", "secretKey" or undefined, as it might be possible to have other kinds of keystore

Reply:
👍